feature_engine/parameter_checks.py

```diff
@@ -34,3 +34,52 @@ def _define_numerical_dict(dict_: Optional[dict]) -> Optional[dict]:
         raise TypeError("The parameter can only take a dictionary or None")
 
     return dict_
+
+
+def _parse_features_to_extract(feats_requested, feats_supported: List[str]) -> Any:
+    """
```

Comment:
in this script, we include functions that check the parameters entered in the __init__ method of the classes. Not sure if this is the intention of this function.

Reply:
That function does indeed check the argument 'features_to_extract' of the ExtractDateFeatures.__init__ method. <br> All those checks were originally within the class init method itself but I thought it better moved somewhere else, also because a future time transformer or something would probably be able to call that function too <br>
Is it fine or am I still missing something?

Reply:
I prefer having the checks in the init itself. It is more readable, than if we have to navigate several scripts. I would take this out of the init and make a function only if this function was to be used by more than 1 class.

See for example all the checks that we have in init params [here](https://github.com/feature-engine/feature_engine/blob/main/feature_engine/selection/drop_psi_features.py#L234)

Reply:
Okay, fine, if anything we could always refactor those checks out of the __init__ in case a new transformer needs them too